File: Distance_Between_Bus_Stops.java
class Solution {
    public int distanceBetweenBusStops(int[] distance, int start, int destination) {
        int ans=0,n=distance.length;
        int i=start;
        int j=destination;
        while(i!=j){
            ans+=distance[i];
            i=(i+1)%n;//To ge to the next elements for clockwise is given (i+1)/n.
        }
        int ans2=0;
        i=start;j=destination;
        while(i!=j){
            i--;
            if(i<0)
                i=n-1;//Now to go anticlockwise , its simple just --i , and when i==-1 just rest it to n-1.
            ans2+=distance[i];
        }
      //now return which path takes minimum distance.
        return Math.min(ans,ans2);
    }
}

File: Distance_Between_Bus_Stops.cpp
class Solution {
public:
    int distanceBetweenBusStops(vector<int>& distance, int start, int destination) {
        int s1=0,s2=0;
        int beg = min(start,destination);
        int end = max(start,destination);
        for(int i=beg;i<end;i++){
            s1+=distance[i];
            distance[i]=0;
        }
        for(auto x:distance){
            s2+=x;
            if(s2>s1) return s1;
        }
        return s2;
    }
};
